src/NFe/Danfe.php

```diff
@@ -1613,9 +1613,9 @@ protected function pGetTextoFatura()
             if (isset($fat)) {
                 $textoIndPag="";
                 $indPag = $this->pSimpleGetValue($this->ide, "indPag");
-                if ($indPag == 0) {
+                if ($indPag === 0) {
```

Comment:
@andrezipf94 dado que nodeValue armazena uma string [1] e `pSimpleGetValue` concatena com os parâmetros `$extraTextBefore` e `$extraTextAfter` que tem como valor default string forçando ser string o retorno.
Agora não deve ser feito o type cast do retorno de `pSimpleGetValue` Ex.: `$indPag = (int) $this->pSimpleGetValue($this->ide, "indPag");` **ou** alterar as condicionais `if` para comparar com string em vez de inteiro?

[1] http://php.net/manual/en/class.domnode.php#domnode.props.nodevalue

Reply:
Correto, não havia me atentado.

Fazer o type cast não resolveria a situação, pois (int) "" resulta em 0, caindo então no mesmo condicional.

Alterei para verificar a tipagem, bem como o valor, só que em String.